rust-runtime/aws-smithy-types/external-types.toml

```diff
@@ -1,2 +1,15 @@
 allowed_external_types = [
+    "bytes::bytes::Bytes",
+    "bytes::buf::buf_impl::Buf",
+
+    # TODO(https://github.com/awslabs/smithy-rs/issues/3033): Feature gate based on unstable versions
+    "http_body::Body",
+    "http_body::combinators::box_body::BoxBody",
+    "hyper::body::body::Body",
```

Comment:
What requires exposing this?

Reply:
- `impl http_body::Body for SdkBody` (src/body.rs:269:1)
- `pub type BoxBody = http_body::combinators::BoxBody<Bytes, Error>;` (src/body.rs:52:1)
- `impl From<hyper::Body> for SdkBody` (src/byte_stream.rs:471)

Reply:
How is BoxBody used? Can we get rid of it?

Reply:
`smithy-checksums` uses it, and this needs to be feature-gated as well. Basically the same as https://github.com/awslabs/smithy-rs/pull/3026#discussion_r1351262625.